File: webapp/src/app/service/class/dashboard-data.ts
import { ActorWithNumber } from "./actor-with-number";
import { PropertyWithNumber } from "./property-with-number";

export class DashboardData {
    avgAcceptedRequestsPerLessor: number = -1;
    avgDeniedRequestsPerLessor: number = -1;
    avgAcceptedRequestsPerTenant: number = -1;
    avgDeniedRequestsPerTenant: number = -1;
    lessorsWithMoreApprovedRequests: ActorWithNumber[] = [];
    lessorsWithMoreDeniedRequests: ActorWithNumber[] = [];
    lessorsWithMorePendingRequests: ActorWithNumber[] = [];
    tenantsWithMoreApprovedRequests: ActorWithNumber[] = [];
    tenantsWithMoreDeniedRequests: ActorWithNumber[] = [];
    tenantsWithMorePendingRequests: ActorWithNumber[] = [];
    lessorsWithMoreApprovedRequestsRatio: ActorWithNumber[] = [];
    lessorsWithLessApprovedRequestsRatio: ActorWithNumber[] = [];
    tenantsWithMoreApprovedRequestsRatio: ActorWithNumber[] = [];
    tenantsWithLessApprovedRequestsRatio: ActorWithNumber[] = [];
    propertiesWithLessAudits: PropertyWithNumber[] = [];
    propertiesWithMoreAudits: PropertyWithNumber[] = [];
    avgAuditsPerProperty: number = -1;
    actorWithLessSocialIdentities: ActorWithNumber[] = [];
    actorWithMoreSocialIdentities: ActorWithNumber[] = [];
    avgSocialIdentitiesPerActor: number = -1;
    tenantWithLessInvoices: ActorWithNumber[] = [];
    tenantWithMoreInvoices: ActorWithNumber[] = [];
    avgInvoicesPerTenant: number = -1;
    totalMoneyDueToSystem: number = -1;
    avgRequestsPerPropertyWithAudits: number = -1;
    avgRequestsPerPropertyWithoutAudits: number = -1;
}

File: src/main/java/com/acme/bnb/controlers/clases/DashboardData.java
package com.acme.bnb.controlers.clases;

import java.util.List;
import lombok.Data;

@Data
public class DashboardData {
    private double avgAcceptedRequestsPerLessor, avgDeniedRequestsPerLessor;
    private double avgAcceptedRequestsPerTenant, avgDeniedRequestsPerTenant;
    private List<ActorWithNumber> lessorsWithMoreApprovedRequests,  lessorsWithMoreDeniedRequests, lessorsWithMorePendingRequests;
    private List<ActorWithNumber> tenantsWithMoreApprovedRequests,  tenantsWithMoreDeniedRequests, tenantsWithMorePendingRequests;
    private List<ActorWithNumber> lessorsWithMoreApprovedRequestsRatio, lessorsWithLessApprovedRequestsRatio;
    private List<ActorWithNumber> tenantsWithMoreApprovedRequestsRatio, tenantsWithLessApprovedRequestsRatio;
    private List<PropertyWithNumber> propertiesWithLessAudits, propertiesWithMoreAudits;
    private double avgAuditsPerProperty;
    private List<ActorWithNumber> actorWithLessSocialIdentities,  actorWithMoreSocialIdentities;
    private double avgSocialIdentitiesPerActor;
    private List<ActorWithNumber> tenantWithLessInvoices,  tenantWithMoreInvoices;
    private double avgInvoicesPerTenant;
    private double totalMoneyDueToSystem;
    private double avgRequestsPerPropertyWithAudits, avgRequestsPerPropertyWithoutAudits;
}
